sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -899,8 +914,9 @@ scatter_rgba(AccessorT acc, simd<uint32_t, N> offsets,
   // TODO (performance) use hardware-supported scale once BE supports it
   constexpr uint32_t Scale = 0;
   const auto SI = get_surface_index(acc);
+  auto loc_offsets = convert<uint32_t>(offsets);
```

Comment:
Don't you need same type of asserts as you added in other places:
```
#ifdef __SYCL_DEVICE_ONLY__
  static_assert(sizeof(Toffset) <= 4, "Unsupported offset type");
#endif
```

Reply:
That would add an error for cases we didn't error before, which I didn't want to do. I only wanted to preserve errors that already exist. Some cases already work because it converts.